iree/hal/cuda/graph_command_buffer.c

```diff
@@ -266,6 +336,24 @@ static uint32_t iree_hal_cuda_splat_pattern(const void* pattern,
   }
 }
 
+static size_t iree_cuda_get_dep_nodes(
```

Comment:
it'd be nice to keep these up near the other functions mutating this state and add some comments about the preconditions/assumptions (being able to quickly see how iree_cuda_add_leaf_node and iree_cuda_flush_dep_nodes relate is useful for readability)